frontend/src/api/github.ts

```diff
@@ -20,24 +20,73 @@ export const isGitHubErrorReponse = <T extends object | Array<unknown>>(
 ): data is GitHubErrorReponse =>
   !!data && "message" in data && data.message !== undefined;
 
+export const retrieveGitHubAppInstallations = async (
+  token: string,
+): Promise<string[]> => {
+  const installationsUrl = "https://api.github.com/user/installations";
+
+  const installationsResponse = await fetch(installationsUrl, {
+    headers: generateGitHubAPIHeaders(token),
+  });
+
+  if (!installationsResponse.ok) {
+    throw new Error(
+      `Failed to fetch installations: ${installationsResponse.statusText}`,
+    );
+  }
+
+  const installationsData = await installationsResponse.json();
+  const installationIds = installationsData.installations.map(
+    (installation: { id: number }) => installation.id,
+  );
+
+  return installationIds;
+};
+
 /**
  * Given a GitHub token, retrieves the repositories of the authenticated user
  * @param token The GitHub token
  * @returns A list of repositories or an error response
  */
 export const retrieveGitHubUserRepositories = async (
   token: string,
+  installationId: string,
   page = 1,
   per_page = 30,
 ): Promise<Response> => {
-  const url = new URL("https://api.github.com/user/repos");
-  url.searchParams.append("sort", "pushed"); // sort by most recently pushed
-  url.searchParams.append("page", page.toString());
-  url.searchParams.append("per_page", per_page.toString());
+  const reposUrl = new URL(
+    `https://api.github.com/user/installations/${installationId}/repositories`,
+  );
```

Comment:
I don't think we can do this unfortunately--we still need the OSS to be able to get a list of repos using a user token

Maybe we put a `/api/github/repositories` endpoint in the backend, and use a flag to decide which API to call